ui/hooks/useMetaMetrics.js

```diff
@@ -0,0 +1,46 @@
+import { useSelector } from 'react-redux';
+import { useRouteMatch } from 'react-router-dom';
+import { PATH_NAME_MAP } from '../helpers/constants/routes';
+import { txDataSelector } from '../selectors';
+
+const PATHS_TO_CHECK = Object.keys(PATH_NAME_MAP);
+
+/**
+ * Returns the current page if it matches our route map, as well as the origin
+ * if there is a confirmation that was triggered by a dapp. These values are
+ * not required but add valuable context to events, and should be included in
+ * the context object on the event payload.
+ *
+ * @returns {{
+ *  page?: MetaMetricsPageObject
+ *  referrer?: MetaMetricsReferrerObject
+ * }}
+ */
+export function useMetaMetrics() {
```

Comment:
[REQ] This name doesn't match the outcome, or rather it gives the impression that this in some way is "using" MetaMetrics.... after talking with a few others, lets use `useSegmentContext` as it is the best name for the outcome.